test/issue-777.xspec

```diff
@@ -3,15 +3,15 @@
 	xmlns:my="my-namespace" xmlns:x="http://www.jenitennison.com/xslt/xspec"
 	xmlns:xs="http://www.w3.org/2001/XMLSchema">
 
-	<x:param as="document-node(element(test))" href="issue-777.xml" name="my:test" />
+	<x:variable as="document-node(element(test))" href="issue-777.xml" name="my:test" />
```

Comment:
Given that #777 was about global `x:param` rather than `x:variable`, did you consider updating this file merely by removing the `query` and `query-at` attributes, instead of changing the global parameter into a global XSpec variable? Global parameters are documented as interacting with the SUT (as opposed to holding test-specific data), so it would be nice to keep testing that the interaction with the SUT doesn't make the bug in #777 return.

(I have more files to look at another day, so this is just a single comment for now.)

Reply:
Thanks @galtm for looking into this.

- Testing #777 needs to be done on both XSLT and XQuery, because their compilers have different codes for creating variable declarations.
- The bug #777 was caused by the way XSpec creates variable declarations in general. Testing with either `x:param` or `x:variable` would do. But testing with both would be even better.

So I made some changes in d26e82bfa3325985359c92f756e9f4ddcd1f040a:
- Restored the test for `x:param` while keeping the same test for `x:variable`
- Removed the `$my:test should be an external document...` scenario, because it wasn't a scope of #777 and I'm afraid it diverts our attention from what's being tested.<br/>(Interaction between SUT and `x:param` or `x:variable` is tested by [test/end-to-end/cases/global-override.xspec](https://github.com/xspec/xspec/blob/master/test/end-to-end/cases/global-override.xspec) and [test/end-to-end/cases/external_global-override.xspec](https://github.com/xspec/xspec/blob/master/test/end-to-end/cases/external_global-override.xspec).)